hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -348,8 +353,13 @@ private InternalScanner postCompactScannerOpen(CompactionRequestImpl request, Sc
         smallestReadPoint = Math.min(fd.minSeqIdToKeep, smallestReadPoint);
         cleanSeqId = true;
       }
+      if (writer != null){
+        LOG.warn("Writer exists when it should not: " + getCompactionTargets().stream()
+          .map(n -> n.toString())
+          .collect(Collectors.joining(", ", "{ ", " }")));
```

Comment:
This is a straight-up codebug, right? If we happen to have a non-null writer here, what's the implication on correctness?

Reply:
That I missed a point where writer reset should have been added or code was changed that allowed this to happen. As far as I can tell this can not happen now and it would be harmless even if it could happen. The worst case scenario is that the cleaner would skip this file once because it would assume the compaction is still ongoing. But it felt wrong not to log a warn before overwriting the writer.

Reply:
> the worst case scenario is that the cleaner would skip this file once because it would assume the compaction is still ongoing

Cool, yeah, agree that is not the end of the world. `LOG.warn` seems appropriate :)